src/basic/alloc-util.h

```diff
@@ -184,17 +185,33 @@ void* greedy_realloc0(void **p, size_t need, size_t size);
 #  define msan_unpoison(r, s)
 #endif
 
-/* This returns the number of usable bytes in a malloc()ed region as per malloc_usable_size(), in a way that
- * is compatible with _FORTIFY_SOURCES. If _FORTIFY_SOURCES is used many memory operations will take the
- * object size as returned by __builtin_object_size() into account. Hence, let's return the smaller size of
- * malloc_usable_size() and __builtin_object_size() here, so that we definitely operate in safe territory by
- * both the compiler's and libc's standards. Note that __builtin_object_size() evaluates to SIZE_MAX if the
- * size cannot be determined, hence the MIN() expression should be safe with dynamically sized memory,
- * too. Moreover, when NULL is passed malloc_usable_size() is documented to return zero, and
- * __builtin_object_size() returns SIZE_MAX too, hence we also return a sensible value of 0 in this corner
- * case. */
-#define MALLOC_SIZEOF_SAFE(x) \
-        MIN(malloc_usable_size(x), __builtin_object_size(x, 0))
+void *expand_to_usable(void *p, size_t newsize) _malloc_ _alloc_(2) _returns_nonnull_;
```

Comment:
hmm, is `_malloc_` really the right thing to use here? reading the docs it suggests that the returned pointer won't alias any other pointer. But that's not the case here, the returned pointer *definitely* will alias the input pointer, after all.

am i reading the docs wrong? are the docs wrong? or is this code wrong? one of this cases it must be

Reply:
Yeah, `_malloc_` is wrong.